docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md

```diff
@@ -69,6 +69,27 @@ Since v7 is a major release, it contains some changes that affect the public API
 These changes were done for consistency, improve stability and make room for new features.
 Below are described the steps you need to make to migrate from v6 to v7.
 
+## XXX: To classify ##
+
+ - type `GridPinnedColumns` => `GridPinnedColumnFields`, type `GridPinnedPosition` => `GridPinnedColumnPosition`
+   - column pinning api signature changes due to the changes above, but data format stays the same
+   - props `pinnedColumns`, `onPinnedColumnsChange`
+ - `.Mui-hovered` => `:hover`
+ - `.pinnedColumns-(left|right)` removed
+ - `getRootDimensions()` now non-null
+ - slot `GridColumnHeaders`, props removed: columnPositions, densityFactor, minColumnIndex
+ - slot `GridRow`, props removed: containerWidth, position
+ - api `mainElementRef` non-null
+ - api `rootElementRef` non-null
+ - api `virtualScrollerRef` non-null
+ - event `renderedRowsIntervalChange`: params changed `GridRenderedRowsIntervalChangeParams` to `GridRenderContext`
+ - type `GridRenderedRowsIntervalChangeParams` removed
+ - type `GridRootContainerRef` removed
+ - slot `GridRow` has typed props now
+ - slot `headerFilterCell`, props removed: filterOperators
+ - DOM changes
```

Comment:
I've gone through the PR, this is the list of breaking changes I found.

In theory no change to the slot props is breaking because they're all typed as `any`, but in practice any change there is a breaking change, it's just that people won't be able to know because there are no types. This is one reason why I think we should absolutely type the slot props.